manifests/config.pp

```diff
@@ -116,16 +116,40 @@
     }
   }
 
-  if ($pam::faillock_log_dir) {
-    file { $pam::faillock_log_dir:
-      ensure   => 'dir',
-      owner    => 'root',
-      group    => 'root',
-      mode     => '0750',
-      seluser  => 'system_u',
-      selrole  => 'object_r',
-      seltype  => 'faillog_t',
-      selrange => 's0',
+  # EL 7 doesn't utilize faillock.conf
+  if ($facts['os']['release']['major'] > '7') and ($pam::manage_faillock_conf) {
```

Comment:
This was the bit that I was referring to when I said something about using hiera defaults.  This comparison probably won't do what you want on Amazon Linux, for example.